### ISSUE TITLE
A way to back/sponsor the project

### ISSUE THREAD

**Issue by xxczaki** [bitinn/node-fetch#680]
## Different options

We can obviously use these 2:

- [Github Sponsors](https://github.com/sponsors) (when available for everyone)
- [Patreon](https://www.patreon.com/)

The problem is, these 2 platforms are designed more for personal accounts. So when you have multiple maintainers, every person will have a different Patreon/GH Sponsors page.

When companies want to support a project, they would rather use something like [Open Collective](https://opencollective.com/) , because:

- their logo and/or name will be visible to other people
- they will what their money is used for (expenses)
- they will be able to better handle the donations (especially the bigger ones) legally
- **they will know, that the money they donated is available to all project maintainers**

However, some companies don't want to support a project just because they are using it or to ensure that it stays maintained. They want to receive a priority/private support to make sure that, for example, security issues get fixed quicker. That's why [Tidelift](https://tidelift.com/about/lifter) was made.

## Advertising

We can encourage people/companies to support node-fetch, by:

- Using a Github Sponsor button and linking Patreon pages or Open Collective
- Using README:
<p align="left">
  <a href="https://opencollective.com/" target="_blank">
    <img src="https://cdn-std.dprcdn.net/files/acc_649651/Q5nVhT" height="80" alt="Open Collective">
  </a>
  <a href="https://www.patreon.com/" target="_blank">
    <img src="https://cdn-std.dprcdn.net/files/acc_649651/plrSCT" height="80" alt="Patreon">
  </a>
</p>

- Using post-installation messages:

![](https://imgur.com/Y4NrV9W.jpg)

 > Small note to the last option: a lot of people don't like seeing any type of advertising when installing a package, especially after the [recent drama](https://www.zdnet.com/article/npm-bans-terminal-ads/).

---

@bitinn Let me know what you think ;)

**Comment by jimmywarting** [comment]
if anything then github sponsor imo.
don't want to clutter the package.json with unrelated stuff/dependencies. that don't belong there.

it's possible to add a `.github/FUNDING.yml` that links to patreon and other stuff

**Comment by Richienb** [comment]
Since GitHub sponsors are still in beta, I'd say we go for opencollective or tidelift.

**Comment by bitinn** [comment]
Quick thoughts:

- Open Collective seems to have more adoption? If it’s being more wildly used, we can go with it (I don’t know its model, or what it requires to distribute payout, yet)

- The good thing about Patreon: it’s personal, you establish a relationship with people, with or without node-fetch. I tend to value them over corporate support (which is of course larger in sum, but can easily change due to business focus or management team fluctuations)

- Of course we want GitHub sponsors, but it’s likely payout to individuals, not teams. We will see when it’s available to us.

- In short, use both FUNDING.yml (GitHub sponsor button) and top of Readme seems good. Will look into Open Collective (someone should try to signup and see what it needs). Could use Patreon for now.

- Also I have doubts on getting corporate support on what’s essentially a spec-compliant polyfill, from my limited talk with some companies, they would rather roll a fork because it’s easy to do (shrug).

(From my phone)

**Comment by bitinn** [comment]
- by Patreon I really mean individual Patreon, they have a Team offering (Premium), but it seems the payout is set by Team Lead. I haven't used that yet.

- Open Collective doesn't seem to have a Team option, so we get the money as a sum and then have to re-distribute using PayPal?

**Comment by Richienb** [comment]
@bitinn If we're planning to use Patreon Premium, we need to make at least $300 a month otherwise we would be essentially losing money. https://www.patreon.com/product/pricing

The problem with Open Collective is that it doesn't seem to have batteries-included teams support.

Liberapay *almost* makes the cut except it doesn't have a donors badge list like in Open Collective.

**Comment by bitinn** [comment]
So the minimum is a bottomline ? We will be charged at least $300 per month no matter what? Seems like a bad business decision to require that for a Team account, I would rather pay a lower service fee.

If so, let's aim to support individual Patreon account for now. (open one if you don't already have them), Open Collective should come once we figure out the payout solution.

(I prefer not to manage payout, I am not in the open-source-sponsor-us business, at least not yet!)

**Comment by Richienb** [comment]
@bitinn Then again, [Liberapay](https://liberapay.com) seems to have teams support and is supported by GitHub so we could use that instead...

**Comment by xxczaki** [comment]
@bitinn On Open Collective you can set core contributors, which are basically like team members :smile:

**Comment by xxczaki** [comment]
In terms of payout: core contributors will be able to submit & approve payouts, so you won’t need to manage them (see [this page](https://docs.opencollective.com/help/expenses))

**Comment by bitinn** [comment]
@xxczaki That seems good to me, I will try to open an account later (real busy lately).

**Comment by bitinn** [comment]
So, I tried to setup this Open Collective thing today, and I realized it requires Stripe account to accept financial support, and Stripe isn't available in Mainland China (can't activate my account).

I don't know if it's possible to link with existing Fiscal Host and still payout correctly, someone need to guide us on that.

- https://docs.opencollective.com/help/collectives/add-or-change-fiscal-host
- https://docs.opencollective.com/help/hosts/create-a-fiscal-host

(This is truly my worry with GitHub sponsor as well, unless they can pay directly to PayPal, I am pretty much not going to be able to use it.)

**Comment by Richienb** [comment]
@bitinn The Wordpress foundation seems to have a 0% commission rate: https://opencollective.com/wpf so we could try that.

**Comment by bitinn** [comment]
Using existing Fiscal Host means every single payout will need to be proven by them. Basically we will be submitting **expense report** to the host's admin. That's not a very flexible setup unless we know them personally.

**Comment by bitinn** [comment]
But of course someone other than me *could* step in and be the team admin and setup a Fiscal Host (wink wink), quite a bit of work if you are not already Stripe member, but that's the cost of getting money from people.

We could also just use individual patron account for now, `FUNDING.yml` should allow us to list many accounts, not just one :)

**Comment by Richienb** [comment]
@bitinn In that case, I'll set up the fiscal host.

**Comment by bitinn** [comment]
@Richienb you have my blessing:

- Make sure you or whoever represent you, meet the legal requirement.
- And please do raise questions when you see them, there are decisions you might regret and can't back out of, if you don't read and consider them carefully.

Just an old man speaking from experience...

**Comment by Richienb** [comment]
The open collective is up! @bitinn @xxczaki I'll need your email addresses as well as anyone else that needs to be team admin.

@bitinn To show the sponsor button, please enable Sponsorships in the [repo settings](https://github.com/bitinn/node-fetch/settings).

**Comment by xxczaki** [comment]
@Richienb My email address is: `xxczaki@pm.me` :smile:

**Comment by bitinn** [comment]
@Richienb good work, mine is `bitinn@gmail.com`

I assume you can add others as team members right?

**Comment by Richienb** [comment]
Yes I can.

**Comment by bitinn** [comment]
@jimmywarting @gr2m please let us know if you want to be added to the list below :)

https://opencollective.com/node-fetch

**Comment by jimmywarting** [comment]
sure, my email is jimmy@warting.se

**Comment by gr2m** [comment]
sure: gregor@martynus.net

**Comment by xxczaki** [comment]
@gr2m done :smile:

**Comment by Richienb** [comment]
Since we've finished setting up Opencollective, this issue can be safely closed unless needed in the future.
